src/controllers/accounts/accounts.ts

```diff
@@ -216,6 +216,18 @@ export class AccountsController extends EventEmitter {
     await this.#storage.set('accounts', this.accounts)
   }
 
+  async reorderAccounts(reordered: Account[]) {
+    const reorderedAddrSet = new Set(reordered.map((acc) => getAddress(acc.addr)))
+
+    const remainingAccounts = this.accounts.filter(
+      (acc) => !reorderedAddrSet.has(getAddress(acc.addr))
+    )
+
+    this.accounts = getUniqueAccountsArray([...reordered, ...remainingAccounts])
+    this.emitUpdate()
+    await this.#storage.set('accounts', this.accounts)
+  }
```

Comment:
I don't think it's necessary to call `getAddress` for accounts that are stored in the extension- their addresses should already be checksummed and even if they are not for whatever reason they will remain lowercase and won't affect the order.